lib/api/api.go

```diff
@@ -1752,12 +1752,21 @@ func checkExpiry(cert tls.Certificate) error {
 		}
 	}
 
-	if leaf.Subject.String() != leaf.Issuer.String() ||
-		len(leaf.DNSNames) != 0 || len(leaf.IPAddresses) != 0 {
-		// The certificate is not self signed, or has DNS/IP attributes we don't
+	if leaf.Subject.String() != leaf.Issuer.String() || len(leaf.IPAddresses) != 0 {
+		// The certificate is not self signed, or has IP attributes we don't
 		// add, so we leave it alone.
 		return nil
 	}
+	if len(leaf.DNSNames) > 1 {
+		// The certificate has more DNS SANs attributes than we ever add, so
+		// we leave it alone.
+		return nil
+	}
+	if len(leaf.DNSNames) == 1 && leaf.DNSNames[0] != leaf.Issuer.CommonName {
+		// The one SAN is different from the issuer, so it's not one of our
+		// newer self signed certificates.
+		return nil
+	}
```

Comment:
Seems weird that checkExpiry function doesn't check expiry in so many edge cases, and returns no error... Defo feels deceptive...

Reply:
Well it's a very special-purpose function, only used to check expiry for the specific purpose of whether we should regenerate the certificate. Possibly it's badly named, but nil in all these cases just means "leave the certificate alone, it's either fine or not our business to recreate it".